qa/rpc-tests/mempool_size_limit_more.py

```diff
@@ -0,0 +1,298 @@
+        # Assert that we did not evict any certificate
+        assert_equal(mempoolstatus['bytes-for-cert'], certPartitionStatus)
+        print()
+
+
+        # We choose a feerate that still classify as high: the midpoint between the lower and higher tx fee of the preious step
+        selected_high_feerate = minfee + (maxfee - minfee) / 2
+        mark_logs(f"Filling tx partition with tiny transactions having feerate {selected_high_feerate:1.8f} (high)", self.nodes, DEBUG_MODE, color = 'c')
+        mark_logs("It's ok to fill tx partition over its limit, we only do not want to evict certificates", self.nodes, DEBUG_MODE, color = 'c')
+        last_tx_size = 0
+        usage = self.nodes[0].getmempoolinfo()
+        addr = self.nodes[0].getnewaddress()
+        utxos = self.nodes[0].listunspent()
+        while(int(usage['bytes']) < NODE0_LIMIT_B - last_tx_size):
+            t = utxos.pop(0)
+            inputs = [{"txid" : t["txid"], "vout" : t["vout"]}]
+            outputs = {addr: t["amount"] - Decimal(str(selected_high_feerate))}
+            rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
+            signedtx = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE")
+            try:
+                self.nodes[0].sendrawtransaction(signedtx['hex'])
+            except JSONRPCException as e:
+                errorString = e.error['message']
+                print(errorString)
+                assert(False)
+            new_usage = self.nodes[0].getmempoolinfo()
+            last_tx_size = int(new_usage['bytes-for-tx']) - int(usage['bytes-for-tx'])
+            usage = new_usage
+        mempoolstatus = self.nodes[0].getmempoolinfo()
+        mark_logs(cc('g', "Current mempool state: bytes: ") + f"{mempoolstatus['bytes']}" +
+                  cc('g', " - bytes-for-tx: ") + f"{mempoolstatus['bytes-for-tx']}" +
+                  cc('g', " - bytes-for-cert: ") + f"{mempoolstatus['bytes-for-cert']}", self.nodes, DEBUG_MODE)
+        # Assert that we did not evict any certificate
+        assert_equal(mempoolstatus['bytes-for-cert'], certPartitionStatus)
+        print()
+
+
+        selected_medium_feerate = 0.00015 + (float(minfee) - 0.00015) / 2
+        mark_logs(f"Sending a medium fee transaction (fee = {selected_medium_feerate:1.8f}) will result in a error", self.nodes, DEBUG_MODE, color = 'c')
+        t = utxos.pop(0)
+        inputs = [{"txid" : t["txid"], "vout" : t["vout"]}]
+        outputs = {addr: t["amount"] - Decimal(str(selected_medium_feerate))}
+        rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
+        signedtx = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE")
+        try:
+            self.nodes[0].sendrawtransaction(signedtx['hex'])
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            print(cc('y', errorString))
```

Comment:
Similar to previous comment:

```suggestion
        inputs = [{"txid" : t["txid"], "vout" : t["vout"]}]

        outputs = {addr: t["amount"]} # for initial estimation
        rawtx = self.nodes[0].createrawtransaction(inputs, outputs) # for initial estimation
        signedtx = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE") # for initial estimation
        tx_size = len(signedtx['hex'])//2 # for initial estimation

        outputs = {addr: t["amount"] - Decimal(selected_medium_feerate * tx_size)}
        rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
        signedtx = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE")
        try:
            self.nodes[0].sendrawtransaction(signedtx['hex'], True)
            assert(False)
        except JSONRPCException as e:
            errorString = e.error['message']
            print(cc('y', errorString))
```

Reply:
<see previous answer>